var/spack/repos/builtin/packages/py-maestrowf/package.py

```diff
@@ -38,6 +41,7 @@ class PyMaestrowf(PythonPackage):
 
     depends_on('python@2.7:2.8,3.5:',  type=('build', 'run'))
     depends_on('py-setuptools',        type='build')
+    depends_on('py-poetry',            type=('build'),        when='@1.1.9:')
     depends_on('py-pyyaml@4.2b1:',     type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-pyyaml@4.2:',     type=('build', 'run'))
```
Spack sorts version suffixes differently than pip